deps/exokit-bindings/canvascontext/include/canvas-context.h

```diff
@@ -160,19 +168,28 @@ class CanvasRenderingContext2D : public ObjectWrap {
   virtual ~CanvasRenderingContext2D();
 
 // protected:
+
+  void Configure();
+
   bool live;
   NATIVEwindow *windowHandle;
   GLuint tex;
   sk_sp<GrContext> grContext;
   sk_sp<SkSurface> surface;
   SkPath path;
+  float shadowBlur;
+  float shadowOffsetX;
+  float shadowOffsetY;
   SkPaint strokePaint;
   SkPaint fillPaint;
   SkPaint clearPaint;
   float lineHeight;
   std::string textAlign;
   TextBaseline textBaseline;
   Direction direction;
+  Nan::Persistent<Value> jsShadowColor;
```

Comment:
Would rather not store any JS-level things here. If necessary, we can store the e.g. string value and reconstruct the JS value again later on get.

There are implications to attaching GC lifetime of these to the lifetime of `CanvasRenderingContext2D` that we should avoid.

Reply:
That's fine for shadow color, but unfortunately the other values can be either a string / canvasgradient / canvaspattern.

Reply:
In that case perhaps a struct?